### PR TITLE
Migrate search to YouTube.js

### PR DESCRIPTION
**Pull request by absidue**
# Migrate search to YouTube.js

## Pull Request Type

- [x] Feature Implementation
- [x] Other - Refactor

## Description
This PR migrates the search functionality over to YouTube.js, this also fixes an issue with the number of channel subscribers being displayed as the number of videos. I also added support for showing the channel handles in the search results. Searching with search filters should be as fast as searching without them now, ytsr requires additional requests, whereas YouTube.js encodes them as protobuf and sends them along in a single request (like YouTube does).

Side note:
If you see any base64 string being sent to YouTube or being returned in a response, it's likely to be a base64 encoded protobuf, you can paste them into this website and see what is inside of it https://protobuf-decoder.netlify.app/.

## Screenshots 
before:
channel-before
after:
channel-after

## Testing 
Try various search queries with and without filters. Also search for some live and upcoming videos to make sure they are displayed correctly (you can copy video titles from https://www.youtube.com/@live)

## Desktop

- **OS:** Windows
- **OS Version:** 10
- **FreeTube version:** 0.18.0

### CODE REVIEW

**Comment by ChunkyProgrammer** [comment]
🎉 code looks great, i'll try it out later tonight

**Comment by ChunkyProgrammer** [comment]
Haven't tested yet but might be a good idea to make sure https://github.com/FreeTubeApp/FreeTube/issues/2385 isn't an issue  (we can filter out hashtags if we need to at the moment)

**Comment by absidue** [comment]
Searching `#shorts` makes these warnings show up in the console, however search results do show up, so it's probably something to report upstream so the warnings go away:
warnings

File: _scripts/build.js [hunk 1]
```diff
@@ -62,13 +62,6 @@ const config = {
     './dist/**/*',
     '!dist/web/*',
     '!node_modules/**/*',
-
-    // renderer
-    'node_modules/{miniget,ytsr}/**/*',
```
Comment (ChunkyProgrammer) [inline _scripts/build.js:67]: I think miniget is still a ytdl-core dependency, would that cause any issues here?

Comment (absidue) [reply _scripts/build.js:67]: That shouldn't be a problem, ytdl-core and it's dependencies get bundled into the `renderer.js` file. Webpack leaves any dependency marked as external untouched or replaces it with your replacement. So with ytsr, the `renderer.js` file still had a `require('ytsr')` call in it, even after bundling, which also means that any dependencies of ytsr will need to be available in the `node_modules` folder at runtime, hence the miniget dependency needing to be listed here while we were using ytsr.
That also meant that we were shipping miniget twice, once inside the `node_modules` and in the `renderer.js` file.

TL;DR ytdl-core will still work as miniget is also bundled into the renderer.js file when webpack bundles ytdl-core.

**Review by ChunkyProgrammer — APPROVED** [review]
Lgtm

**Review by PikachuEXE — APPROVED** [review]
teste locally

**Review by efb4f5ff-1298-471a-8973-3d47447115dc — APPROVED** [review]

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Could someone open a PR in the docs repo for this?

**Comment by PikachuEXE** [comment]
@efb4f5ff-1298-471a-8973-3d47447115dc https://github.com/FreeTubeApp/FreeTube-Docs/pull/78
